index.js

```diff
@@ -278,19 +280,45 @@ class GithubScm extends Scm {
             "else echo 'sd-step exec core/git'; fi)";
         const command = [];
 
-        // Git clone
-        command.push(`echo Cloning ${checkoutUrl}, on branch ${config.branch}`);
         command.push('if [ ! -z $SCM_CLONE_TYPE ] && [ $SCM_CLONE_TYPE = ssh ]; ' +
             `then export SCM_URL=${sshCheckoutUrl}; ` +
             'elif [ ! -z $SCM_USERNAME ] && [ ! -z $SCM_ACCESS_TOKEN ]; ' +
             `then export SCM_URL=https://$SCM_USERNAME:$SCM_ACCESS_TOKEN@${checkoutUrl}; ` +
             `else export SCM_URL=https://${checkoutUrl}; fi`);
-        command.push(`${gitWrapper} `
-                + `"git clone --recursive --quiet --progress --branch ${config.branch} `
-                + '$SCM_URL $SD_SOURCE_DIR"');
-        // Reset to SHA
-        command.push(`${gitWrapper} "git reset --hard ${checkoutRef} --"`);
-        command.push(`echo Reset to ${checkoutRef}`);
+
+        if (config.manifest) {
+            const repoDownloadUrl = 'https://storage.googleapis.com/git-repo-downloads/repo';
+            const sdRepoReleasesUrl = 'https://github.com/screwdriver-cd/sd-repo/releases/latest';
+
+            command.push('echo Checking out code using the repo manifest defined in '
+                + `${config.manifest}`);
+
+            // Get the repo binary
+            command.push(`curl ${repoDownloadUrl} > /usr/local/bin/repo`);
+            command.push('chmod a+x /usr/local/bin/repo');
+
+            // // Get the sd-repo binary and execute it
+            command.push(`wget -q -O - ${sdRepoReleasesUrl} | egrep -o `
+                + "'/screwdriver-cd/sd-repo/releases/download/v[0-9.]*/sd-repo_linux_amd64' "
+                + '| wget --base=http://github.com/ -q -i - -O /usr/local/bin/sd-repo');
+            command.push('chmod a+x /usr/local/bin/sd-repo');
+            command.push(`sd-repo -manifestUrl=${config.manifest} `
+                + `-sourceRepo=${config.org}/${config.repo}`);
+
+            // // Export $SD_SOURCE_DIR to source repo path and cd into it
+            command.push('export SD_SOURCE_DIR = $SD_SOURCE_DIR/$(cat sourcePath)');
+            command.push('cd $SD_SOURCE_DIR');
```

Comment:
Is `cd $SD_SOURCE_DIR` necessary in all builds or only for PR?

Reply:
I believe all builds. The user should still expect to being the build at $SD_SOURCE_DIR